src/lib.rs

```diff
@@ -5,17 +5,24 @@ use marker_api::prelude::*;
 use marker_api::{LintPass, LintPassInfo, LintPassInfoBuilder};
 
 mod clippy;
+mod custom;
 
 #[derive(Default)]
 struct MyLintPass {}
 marker_api::export_lint_pass!(MyLintPass);
 
 impl LintPass for MyLintPass {
     fn info(&self) -> LintPassInfo {
-        LintPassInfoBuilder::new(clippy::lints().into_boxed_slice()).build()
+        let mut lints = clippy::lints();
+        lints.append(&mut custom::lints());
```

Comment:
```suggestion
        lints.extend(custom::lints());
```
I generally never find a use case for `append`. When you want to append you'll always don't need to vec that you appended.

Reply:
AFAIK, it's the other way around. `append` has knowledge about the type, and can copy the whole buffer at once. Maybe even with a `memcpy` for copy types. `extend` first turns the collection into an iterator and then moves each item one at a time. `extend` almost acts like inserting each element individually.

Clippy has [clippy::extend_with_drain](https://rust-lang.github.io/rust-clippy/master/index.html#/extend_with_drain) which isn't quite the same, but follows the same logic.

This might optimize to the same thing, but we can help the compiler a bit. The difference might be stronger for data types with more constraints, like a heap. :)

Reply:
Vector uses unstable trait specialization feature, so it should definitely be optimized for the case when the `Vec` is extend with another `Vec` ([code](https://github.com/rust-lang/rust/blob/4f4a413fe6931d0ad9d3ac6bd20ff36398961e64/library/alloc/src/vec/mod.rs#L2823)).